app/master/build_store.py

```diff
@@ -0,0 +1,50 @@
+from collections import OrderedDict
+from itertools import islice
+from typing import List
+
+from app.master.build import Build
+from app.util.exceptions import ItemNotFoundError
+
+
+class BuildStore(object):
+    """
+    Build storage service that stores and handles all builds.
+    """
+    _all_builds_by_id = OrderedDict()
+
+    @classmethod
+    def get(cls, build_id: int) -> Build:
+        """
+        Returns a build by id
+        :param build_id: The id for the build whose status we are getting
+        """
+        build = cls._all_builds_by_id.get(build_id)
+        if build is None:
+            raise ItemNotFoundError('Invalid build id: {}.'.format(build_id))
+
+        return build
+
+    @classmethod
+    def get_range(cls, start: int, end: int) -> List['Build']:
+        """
+        Returns a list of all builds.
+        :param offset: The starting index of the requested build
```

Comment:
The variable names are start and end but the docstrings mention offset and limit